utils/ssh.py

```diff
@@ -480,6 +480,14 @@ def set_initial_file_end(self):
             fstat = sshtail._sftp_client.stat(self._remote_filename)
             self._remote_file_size = fstat.st_size  # Seed initial size of file
 
+    def raw_output(self):
+        """Returns the output of the log as one (raw) string"""
+        return '\n'.join(self.lines_as_list())
```

Comment:
I would do it like this:

     def __iter__(self):
          for line in self.raw_lines:
               yield line.rstrip()

     def raw_lines(self):
         with self as sshtail:
             fstat = sshtail._sftp_client.stat(self._remote_filename)
             if self._remote_file_size is not None:
                 if self._remote_file_size < fstat.st_size:
                     remote_file = self._sftp_client.open(self._remote_filename, 'r')
                     remote_file.seek(self._remote_file_size, 0)
                     while (remote_file.tell() < fstat.st_size):
                         line = remote_file.readline()                                # Note the  missing rstrip() here!
                         yield line
             self._remote_file_size = fstat.st_size

     def raw_string():
          retrun ''.join(l for l in raw_lines())